common-items.lic

```diff
@@ -129,7 +129,7 @@ module DRCI
   end
 
   def count_items_in_container(item, container)
-    contents = DRC.bput("rummage /C #{item} IN MY #{container}", '^You rummage .*', 'That would accomplish nothing')
+    contents = DRC.bput("rummage /C #{item.split.last} IN MY #{container}", '^You rummage .*', 'That would accomplish nothing')
```

Comment:
❓ Did you test this with single-word items and multi-word items?

Reply:
yes, it works fine. this is the same thing used in ;restock, I'm not sure the one in common-items.lic is different.

```You tap a carving knife inside your harvesting satchel.
[theurgy]>rummage /C knife IN MY harvesting satchel
You rummage through a harvesting satchel beaded with tiny purple Ilithic apples crafted from felwood looking for something similar to "knife" and
  see a carving knife.
[theurgy]>tap my shaper
You tap a brass wood shaper inside your harvesting satchel.
[theurgy]>rummage /C shaper IN MY harvesting satchel
>
You rummage through a harvesting satchel beaded with tiny purple Ilithic apples crafted from felwood looking for something similar to "shaper" and
  see a brass wood shaper.
[theurgy]>tap my prayer chain
>
You tap an empty prayer bead chain inside your pink backpack.
[theurgy]>rummage /C chain IN MY pink backpack
You rummage through a bedazzled pink backpack with a rainbow sapphire clasp looking for something similar to "chain" and see an empty prayer bead
  chain.
[theurgy]>rummage /C sage IN MY satchel
You rummage through a harvesting satchel beaded with tiny purple Ilithic apples crafted from felwood looking for something similar to "sage" but
  there is nothing in there like that.
[theurgy]>rummage /C lavender IN MY satchel
You rummage through a harvesting satchel beaded with tiny purple Ilithic apples crafted from felwood looking for something similar to "lavender"
  but there is nothing in there like that.
[theurgy]>count my first incense in my satchel
I could not find what you were referring to.
[theurgy]>rummage /C flint IN MY satchel
You rummage through a harvesting satchel beaded with tiny purple Ilithic apples crafted from felwood looking for something similar to "flint" and
  see a piece of black flint.
[theurgy]>rummage /C wine IN MY satchel
You rummage through a harvesting satchel beaded with tiny purple Ilithic apples crafted from felwood looking for something similar to "wine" but
  there is nothing in there like that.
[theurgy]>rummage /C taffelberries IN MY satchel
>
You rummage through a harvesting satchel beaded with tiny purple Ilithic apples crafted from felwood looking for something similar to
  "taffelberries" but there is nothing in there like that.
[theurgy]>count my first oil in my satchel
There are three parts left of the holy oil.
[theurgy]>count my second oil in my satchel
I could not find what you were referring to.
[theurgy]>pray Tamsine
You roll your eyes heavenward and murmur a soft prayer to Tamsine, asking for patience.
```

Reply:
https://github.com/rpherbig/dr-scripts/blob/master/restock.lic#L95

Reply:
the prayer chain is the multi word item in that log.

Reply:
Huh, the regex in `restock` is also different: `contents.scan(/#{item}/).size`

Would you be up for having `restock` use this helper as part of this PR?

Reply:
Would this be enough?

```diff --git a/restock.lic b/restock.lic
index 29f96cf2..ee07e253 100644
--- a/restock.lic
+++ b/restock.lic
@@ -92,8 +92,7 @@ class Restock
     return 0 if tap_result.nil?

     container = tap_result
-    contents = bput("rummage /C #{item.split.last} IN MY #{container}", '^You rummage .*')
-    contents.scan(/#{item}/).size
+    contents = count_items_in_container(container, item)
   end

   def valid_item_data?(item_data)```

Reply:
would thiis be enough? I can't test it until tomorrow. 
```diff --git a/restock.lic b/restock.lic
index 29f96cf2..ee07e253 100644
--- a/restock.lic
+++ b/restock.lic
@@ -92,8 +92,7 @@ class Restock
     return 0 if tap_result.nil?

     container = tap_result
-    contents = bput("rummage /C #{item.split.last} IN MY #{container}", '^You rummage .*')
-    contents.scan(/#{item}/).size
+    contents = count_items_in_container(container, item)
   end

Reply:
That looks good. I'd like you to test it before I merge, though

Reply:
glad I tested, its count_items_in_container(item, container). Looks good now though

```--- Lich: restock active.
[restock]>count my first crossbow bolts
You count some crossbow bolts in the rucksack and see there are thirty left.
>
[restock]>count my second crossbow bolts
I could not find what you were referring to.
>
[restock]>tap my fresh milk
You tap a bottle of fresh milk inside your oilcloth rucksack.
>
[restock]>rummage /C milk IN MY oilcloth rucksack
You rummage through an oilcloth rucksack looking for something similar to "milk" and see a bottle of fresh milk and a bottle of fresh milk.
>
--- Lich: restock has exited.